src/type/definition.js

```diff
@@ -478,12 +478,12 @@ export type GraphQLIsTypeOfFn = (
   info: GraphQLResolveInfo
 ) => boolean
 
-export type GraphQLFieldResolveFn<TSource> = (
+export type GraphQLFieldResolveFn<TSource, TResult> = (
   source: TSource,
   args: {[argName: string]: mixed},
   context: mixed,
   info: GraphQLResolveInfo
-) => mixed
+) => TResult
```

Comment:
If `What issue would Flow not catch that it does catch after this change?` means what is the usage of `TResult`.
With `TResult` the Type system can check the out type for a resovle, and  ensure it is corresponding to the lower level .There is a example:
[a demonstration for a Relay's Connection constraint](https://github.com/iamchenxin/relayql/blob/master/example/starWarsSchema.js#L168) With the `TResult`, the type system will check the returned data the users output,if the user typo, like 'id' -> 'idd'. flow will give errors.
![untitled](https://cloud.githubusercontent.com/assets/8873545/17104271/5d7e41a0-523f-11e6-9dc0-14e1272afc22.png)